tests/test_differentiation.py

```diff
@@ -65,6 +65,54 @@ def test_adjoint_diff(n_qubits: int, n_layers: int) -> None:
     # )
 
 
+@pytest.mark.parametrize("n_qubits", [3, 5])
+@pytest.mark.parametrize("batch_size", [1, 2])
+@pytest.mark.parametrize("ops_op", [pyq.Z, pyq.Y])
+@pytest.mark.parametrize("dtype", [torch.complex128])
+def test_sampled_diff(
+    n_qubits: int,
+    batch_size: int,
+    ops_op: Primitive,
```

Comment:
```suggestion
    ops_op: type[Primitive],
```

Tbc though.

Reply:
pre-commit does not like it, keeping Primitive for now